Cannot build any project with system-wide DTC

**Describe the bug**
It's not possible to compile even a simple project for several Atmel-based boards if there is a system-wide DTC.

Might be caused by https://github.com/zephyrproject-rtos/zephyr/commit/c109e7691d32c027f5a1dd8e257f0f34b0376523.

**Steps to reproduce the behavior:**
```
1. west init --mr v2.5.0
2. west update
3. export ZEPHYR_TOOLCHAIN_VARIANT=gnuarmemb
4. export GNUARMEMB_TOOLCHAIN_PATH=/home/runner/.platformio/packages/toolchain-gccarmnoneeabi
5. west -v build -p auto -b arduino_zero zephyr/samples/hello_world
```

**Impact**
This blocks release of the latest Zephyr on several dev-platforms within the PlatformIO ecosystem.

**Logs and console output**
```
> west -v build -p auto -b arduino_zero zephyr/samples/hello_world

ZEPHYR_BASE=/home/runner/west/zephyr (origin: configfile)
-- west build: generating a build system
cmake version 3.16.4 is OK; minimum version is 3.13.1
Running CMake: /home/runner/.platformio/packages/tool-cmake/bin/cmake -DWEST_PYTHON=/home/runner/venvs/west/bin/python3 -B/home/runner/west/build -S/home/runner/west/zephyr/samples/hello_world -GNinja -DBOARD=arduino_zero
Including boilerplate (Zephyr base): /home/runner/west/zephyr/cmake/app/boilerplate.cmake
-- Application: /home/runner/west/zephyr/samples/hello_world
-- Zephyr version: 2.5.0 (/home/runner/west/zephyr)
-- Found Python3: /home/runner/venvs/west/bin/python3.9 (found suitable exact version "3.9.1") found components: Interpreter 
-- Found west (found suitable version "0.9.0", minimum required is "0.7.1")
-- Board: arduino_zero
-- Cache files will be written to: /home/runner/.cache/zephyr
-- Found dtc: /home/runner/.platformio/packages/tool-dtc/dtc (found suitable version "1.4.7", minimum required is "1.4.6")
-- Found toolchain: gnuarmemb (/home/runner/.platformio/packages/toolchain-gccarmnoneeabi)
-- Found BOARD.dts: /home/runner/west/zephyr/boards/arm/arduino_zero/arduino_zero.dts
FATAL ERROR: Unrecognized check name "unique_unit_address_if_enabled"
CMake Error at /home/runner/west/zephyr/cmake/dts.cmake:205 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /home/runner/west/zephyr/cmake/app/boilerplate.cmake:535 (include)
  /home/runner/west/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:24 (include)
  /home/runner/west/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:35 (include_boilerplate)
  CMakeLists.txt:5 (find_package)

-- Configuring incomplete, errors occurred!
FATAL ERROR: command exited with status 1: /home/runner/.platformio/packages/tool-cmake/bin/cmake -DWEST_PYTHON=/home/runner/venvs/west/bin/python3 -B/home/runner/west/build -S/home/runner/west/zephyr/samples/hello_world -GNinja -DBOARD=arduino_zero

```
**Environment (please complete the following information):**
 - OS: Any (Linux, MacOS, Windows)
 - Toolchain: GCC 8.2.1
 - Zephyr v2.5.0
 - west v0.9.0

--- a/cmake/dts.cmake
+++ b/cmake/dts.cmake
@@ -188,6 +188,14 @@ if(SUPPORTS_DTS)
   if (check)
     set(DTC_NO_WARN_UNIT_ADDR "-Wno-unique_unit_address")
   endif()
+  set(VALID_EXTRA_DTC_FLAGS "")
+  foreach(extra_opt ${EXTRA_DTC_FLAGS})
+    check_dtc_flag(${extra_opt} check)
+    if (check)
+      list(APPEND VALID_EXTRA_DTC_FLAGS ${extra_opt})
+    endif()
+  endforeach()
+  set(EXTRA_DTC_FLAGS ${VALID_EXTRA_DTC_FLAGS})
   execute_process(
     COMMAND ${DTC}
     -O dts

